common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -3015,6 +3016,39 @@ def ensure_indexes(self):
         """
         self.db_connection.ensure_indexes()
 
+    def _get_block_usage_locator(self, course_key, block_key):
+        return BlockUsageLocator(course_key, block_key.type, block_key.id)
+
+    def _format_course_structure(self, course_key, structure):
+        # Convert the root key to a locator string
+        root = unicode(self._get_block_usage_locator(course_key, structure[u'root']))
+        blocks = {}
+
+        for key, block in structure[u'blocks'].iteritems():
+            locator = unicode(self._get_block_usage_locator(course_key, key))
+            fields = block.get(u'fields', {})
+            children = fields.get(u'children', [])
+            children = [unicode(self._get_block_usage_locator(course_key, child)) for child in children]
+
+            blocks[locator] = Block(locator,
+                                    block[u'block_type'],
+                                    display_name=fields.get(u'display_name', None),
+                                    format=fields.get(u'format', None),
+                                    graded=fields.get(u'graded', False),
+                                    children=children)
+
+        return CourseStructure(root, blocks)
+
+    def get_course_structure(self, course_id, version=None):
```

Comment:
I wonder if it would be too convoluted to have a function arg to the thing building the `CourseStructures` so that it could call XBlock class methods on fields, and return a bit more control to the XBlock layer? So by default, the structure that's returned has no field information, and the Blocks contain only `id`, `block`, and `children`. If you want extra fields attached to your block. Does that make any sense?